lib/remote_storage/s3.rb

```diff
@@ -16,6 +16,8 @@ def format_etag(etag)
     end
 
     def do_put_request(url, data, content_type)
+      validate_content_type(content_type)
```

Comment:
Why only do this for a single adapter, and not in the place where we already do other validations?

https://github.com/5apps/liquor-cabinet/blob/71d138894ebae6ee5457794e8ef0a728da37d29c/lib/remote_storage/rest_provider.rb#L122

Reply:
By the way, I think it would be much cleaner to not halt in the validation function, but rather to do something like this, where it's used:

```rb
halt 415 unless valid_content_type?(str)
```

Reply:
Thanks, I've done that in c2d5fc9